packages/next/src/server/app-render/create-component-tree.tsx

```diff
@@ -669,8 +669,13 @@ async function createComponentTreeInternal({
     return [
       actualSegment,
       <React.Fragment key={cacheNodeKey}>
-        {metadata}
         {pageElement}
+        {/*
+         * The order here matters since a parent might call findDOMNode().
+         * findDOMNode() will return the first child if multiple children are rendered.
+         * But React will hoist metadata into <head> which breaks scroll handling.
+         */}
+        {metadata}
```

Comment:
With [Fragment Refs](https://github.com/reactjs/rfcs/blob/fragmentrefs-rfc/text/0000-fragment-refs.md), we wouldn't have this problem. We'd just iterate over the children and skip over metadata.